From cb4e75c39477967daa11caba6390db0f5e305add Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Thu, 16 Mar 2023 10:42:56 +0800
Subject: [PATCH] [riscv] Remove the unused kOffHeapTrampolineRegister

Port commit 0e1e2b838a8f9ecc61667f083041417a5c19345b

Change-Id: Ibfce37eff4e4540a01d5c707ac32702acc082b73
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4344114
Auto-Submit: Yahan Lu <yahan@iscas.ac.cn>
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#86475}
---
 src/codegen/riscv/macro-assembler-riscv.cc | 16 ----------------
 src/codegen/riscv/macro-assembler-riscv.h  |  4 ----
 src/codegen/riscv/register-riscv.h         |  1 -
 3 files changed, 21 deletions(-)

diff --git a/src/codegen/riscv/macro-assembler-riscv.cc b/src/codegen/riscv/macro-assembler-riscv.cc
index 7a54469c61..747666b698 100644
--- a/src/codegen/riscv/macro-assembler-riscv.cc
+++ b/src/codegen/riscv/macro-assembler-riscv.cc
@@ -5367,22 +5367,6 @@ void MacroAssembler::JumpToExternalReference(const ExternalReference& builtin,
   Jump(code, RelocInfo::CODE_TARGET, al, zero_reg, Operand(zero_reg));
 }
 
-void MacroAssembler::JumpToOffHeapInstructionStream(Address entry) {
-  // Ld a Address from a constant pool.
-  // Record a value into constant pool.
-  ASM_CODE_COMMENT(this);
-  if (!v8_flags.riscv_constant_pool) {
-    li(kOffHeapTrampolineRegister, Operand(entry, RelocInfo::OFF_HEAP_TARGET));
-  } else {
-    RecordEntry(entry, RelocInfo::OFF_HEAP_TARGET);
-    RecordRelocInfo(RelocInfo::OFF_HEAP_TARGET, entry);
-    auipc(kOffHeapTrampolineRegister, 0);
-    LoadWord(kOffHeapTrampolineRegister,
-             MemOperand(kOffHeapTrampolineRegister, 0));
-  }
-  Jump(kOffHeapTrampolineRegister);
-}
-
 void MacroAssembler::LoadWeakValue(Register out, Register in,
                                    Label* target_if_cleared) {
   ASM_CODE_COMMENT(this);
diff --git a/src/codegen/riscv/macro-assembler-riscv.h b/src/codegen/riscv/macro-assembler-riscv.h
index 8f36921b1c..24de42c7f5 100644
--- a/src/codegen/riscv/macro-assembler-riscv.h
+++ b/src/codegen/riscv/macro-assembler-riscv.h
@@ -1350,10 +1350,6 @@ class V8_EXPORT_PRIVATE MacroAssembler : public MacroAssemblerBase {
   // Jump to the builtin routine.
   void JumpToExternalReference(const ExternalReference& builtin,
                                bool builtin_exit_frame = false);
-
-  // Generates a trampoline to jump to the off-heap instruction stream.
-  void JumpToOffHeapInstructionStream(Address entry);
-
   // ---------------------------------------------------------------------------
   // In-place weak references.
   void LoadWeakValue(Register out, Register in, Label* target_if_cleared);
diff --git a/src/codegen/riscv/register-riscv.h b/src/codegen/riscv/register-riscv.h
index 5f1fc91474..915ecf6a3a 100644
--- a/src/codegen/riscv/register-riscv.h
+++ b/src/codegen/riscv/register-riscv.h
@@ -288,7 +288,6 @@ constexpr Register kJavaScriptCallTargetRegister = kJSFunctionRegister;
 constexpr Register kJavaScriptCallNewTargetRegister = a3;
 constexpr Register kJavaScriptCallExtraArg1Register = a2;
 
-constexpr Register kOffHeapTrampolineRegister = t6;
 constexpr Register kRuntimeCallFunctionRegister = a1;
 constexpr Register kRuntimeCallArgCountRegister = a0;
 constexpr Register kRuntimeCallArgvRegister = a2;
-- 
2.35.1

